Improve persistent storage tests

pmm-sumo opened the pull request:
**Description:** 

This addresses #4563

I was not able to replicate the original issue (I did run the tests in a loop for some time and looked through the code). Perhaps the timeout was too short and the tests suite failed when it was running on an instance that had high resource usage. I made some small improvements to the test (increased the timeout, reduced test logging output)

codeboten commented on exporter/exporterhelper/internal/persistent_storage_test.go:
```diff
@@ -174,6 +172,7 @@ func TestPersistentStorage_CorruptedData(t *testing.T) {
 
 	for _, c := range cases {
 		t.Run(c.name, func(t *testing.T) {
+			ext := createStorageExtension(path)
```
Why this change? I didn't see it mentioned in the description.

pmm-sumo replied:
This essentially moves [this line](https://github.com/open-telemetry/opentelemetry-collector/pull/4592/files#diff-b70537f6db93c86aa3901865578263fe3dd9557c75db043e585de2983135913aL114) here, which makes sure that each run gets a new storage extension instance

tigrannajaryan commented on exporter/exporterhelper/internal/persistent_storage_test.go:
```diff
@@ -217,7 +216,7 @@ func TestPersistentStorage_CorruptedData(t *testing.T) {
 				defer newPs.mu.Unlock()
 				return newPs.size() == c.desiredQueueSize && len(newPs.currentlyDispatchedItems) == c.desiredNumberOfDispatchedItems
-			}, 500*time.Millisecond, 10*time.Millisecond)
+			}, 1000*time.Millisecond, 10*time.Millisecond)
```
If the timeout is the suspect I think you can be more generous with this, e.g. 5 seconds or something like that. It should typically very fast so the larger value of the timeout is not going to affect the normal runs.

pmm-sumo replied:
Good point, I have set timeout to 5 seconds consistently across the test

pmm-sumo commented:
Adjusted the timeout and rebased. I am not sure if it's worth adding changelog entry for such change? (and I don't have permissions to attach the label)

codeboten commented:
> Adjusted the timeout and rebased. I am not sure if it's worth adding changelog entry for such change? (and I don't have permissions to attach the label)

Agreed it doesn't make sense, added the label to skip it.

pmm-sumo commented:
> > Adjusted the timeout and rebased. I am not sure if it's worth adding changelog entry for such change? (and I don't have permissions to attach the label)
> 
> Agreed it doesn't make sense, added the label to skip it.

Thank you!
